### DISCUSSION TITLE
Connection Migration on Preferred Address for ngtcp2

### DISCUSSION THREAD

**Discussion by YifangZhang** [ngtcp2/ngtcp2#1250] [Q&A] [answered]
Dear authors and maintainers,

I am a relatively new user of both QUIC and ngtcp2.

Based on RFC 9000 [Section 9.6](https://datatracker.ietf.org/doc/html/rfc9000#section-9.6), there is a possible migration on the server side to migrate the whole connection into the server's preferred address.

Is it possible for someone to point out this part of the code/example in ngtcp2 package?

I have some brief search in the package and find "ngtcp2_select_preferred_addr" at [this place](https://nghttp2.org/ngtcp2/types.html#c.ngtcp2_select_preferred_addr), but not sure how I can use this to perform the preferred address migration on the server side.

Thanks!

**Comment by tatsuhiro-t** [comment, marked as answer]
There is no server side preferred address related code except for the encoding/decoding of transport parameters.
The simplest server implementation is done outside ngtcp2, that is listen on preferred address separately.  This is what examples/server does.

**Reply by YifangZhang** [reply]
Is there any update since it is 2025 now?
